ext/mysqlnd/mysqlnd_wireprotocol.c

```diff
@@ -292,9 +305,12 @@ mysqlnd_read_packet_header_and_body(MYSQLND_PACKET_HEADER * packet_header,
 {
 	DBG_ENTER("mysqlnd_read_packet_header_and_body");
 	DBG_INF_FMT("buf=%p size=%zu", buf, buf_size);
+	assert(error_info->error_no == 0);
 	if (FAIL == mysqlnd_read_header(pfc, vio, packet_header, stats, error_info)) {
 		SET_CONNECTION_STATE(connection_state, CONN_QUIT_SENT);
-		SET_CLIENT_ERROR(error_info, CR_SERVER_GONE_ERROR, UNKNOWN_SQLSTATE, mysqlnd_server_gone);
+		if (error_info->error_no == 0) {
+			SET_CLIENT_ERROR(error_info, CR_SERVER_GONE_ERROR, UNKNOWN_SQLSTATE, mysqlnd_server_gone);
+		}
```

Comment:
For the same reason, I don't think this check should be done either, but is there any reason why `error_no` has to be `0`?

Reply:
I consider 0 to mean that no error occurred, but this cannot be guaranteed. so may be compare with CLIENT_INTERACTION_TIMEOUT here?

Reply:
I tried various things.
There may already be a meaningful error set here, so checking if it's `0` makes sense to me.

Reply:
Got it.